api/app.py

```diff
@@ -0,0 +1,27 @@
+import os
+from flask import Flask
+
+
+def create_app(config):
+    app = Flask(__name__)
+    # Load given stage config
+    app.config.from_object(config)
+    # Override with user config, if exists
+    app.config.from_envvar('USER_CONFIG', silent=True)
+    # Register blueprints within app context
+    with app.app_context():
+        #from errors import blueprint_errors
+        # app.register_blueprint(blueprint_errors)
+        from api import blueprint_api
+        app.register_blueprint(blueprint_api, url_prefix='/api')
```

Comment:
Har url_prefix här rätt att skriva över prefixet som vi anger när vi skapar blueprinten?

+blueprint_api = Blueprint('api', __name__) ? 

Nu har vi samma på båda ställena,

Reply:
Andra argumentet i Blueprint-konstruktorn är namnet för pythonpaketet, inte url_prefix. Du skulle kunna ange ett `url_prefix` också i blueprintkonstruktorn, e.g. `Blueprint('api', __name__, url_prefix='api')` och i det fallet vet jag inte riktigt vad som händer. Kanske blir överskrivet av `register_blueprint` eller så konkateneras prefixen. 

Dock tydligare att definiera när man väl lägger registrerar blueprintet, tycker jag. Då har man alla prefixen samlade på ett ställe, dvs i app.py.

Reply:
Aha, dåså. Håller med om att det är mycket tydligare att montera blueprinten från applikationen.

Jag tolkade följande:

" When you bind a function with the help of the @simple_page.route decorator the blueprint will record the intention of registering the function show on the application when it’s later registered. Additionally it will prefix the endpoint of the function with the name of the blueprint which was given to the Blueprint constructor (in this case also simple_page)."

Från: http://flask.pocoo.org/docs/0.12/blueprints/

Som att endpointen blev prefixad med blueprint-namnet, men dom syftade nog på något annat. 

Hur som helst håller jag med om att det är tydligare att montera blueprinten från applikationen än att ha den infon gömd inne i själv blurprinten.